drivers/dht/dht.c

```diff
@@ -67,79 +95,118 @@ int dht_init(dht_t *dev, const dht_params_t *params)
 
     /* check parameters and configuration */
     assert(dev && params &&
-           ((dev->type == DHT11) || (dev->type == DHT22) || (dev->type == DHT21)));
+           ((params->type == DHT11) || (params->type == DHT22) || (params->type == DHT21)));
```

Comment:
as you're at it, why not make this an `assert` on its own? Meaning, have one assert to verify the pointers and another to check device type.